core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -560,7 +618,9 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
   override def isExecutorActive(id: String): Boolean = synchronized {
     executorDataMap.contains(id) &&
       !executorsPendingToRemove.contains(id) &&
-      !executorsPendingLossReason.contains(id)
+      !executorsPendingLossReason.contains(id) &&
+      !executorsPendingDecommission.contains(id)
```

Comment:
Here the entire node will be decommissioned , So instead of checking for each executor id contains in executorsPendingDecommission, can we have the filter on the node level with its HostName as the input parameter. So there will be only one entry of the node hostname that would be added in the some decommission tracker.At the end we don't want any task to be scheduled on the executor running on the node. We can check this in org.apache.spark.scheduler.cluster.CoarseGrainedSchedulerBackend.DriverEndpoint#makeOffers

Reply:
We could, but we also use this as a basis for other cluster managers which may not have an entire node decom always (hence executor decommissioning).

Reply:
I am considering this scenario where there is spotloss in aws or Preemptible VM loss in GCP, the entire node will not be available and all the executors on that node will not be available.There may be scenario as per the existing code where the code is checking the decommissioning of executors and then API call is send to update the executorsPendingDecommission for each executors and in mean time when the executor information is added into the executorsPendingDecommission and node is lost/ about to be lost. Since there might be some of the enteries of executors that are not added into executorsPendingDecommission and we are still scheduling a task in those executors.

If it would have been on the node level, just there is need to add first node info in the NodePendingToDecommission and rest if the other executors running same node will try to add there entry can be ignored. In this there is less chance of the task to be scheduled on any executor on the decommissioning nodes. This is possible at the yarn end where AM can send the request to RM to find the updated states of the node. Not sure if its possible in the case of k8s where each executorpods can also get the information about hostName where that pods are running and pods loss reason. 

If its not possible to get the above info in the k8s, can we scope this executor decommission to k8s level and also add the node level decommissioning also here where other cluster manager can take the benefit of node level decommissioning

Reply:
We could also add node-level information when we implement YARN support, I'm not sure what that would give us over just adding the executors on the node to the current structure.